src/nvim/ex_getln.c

```diff
@@ -263,7 +263,10 @@ static uint8_t *command_line_enter(int firstc, long count, int indent)
   s->old_botline = curwin->w_botline;
 
   // set some variables for redrawcmd()
-  ccline.cmdfirstc = (s->firstc == '@' ? 0 : s->firstc);
```

Comment:
The existing shenanigans with `cmdfirstc` is a bit unfortunate... `get_cmdline_type()` should support the new `#` type. Either a boolean flag for this type needs to be added in `ccline`, or perhaps better variable that is the unadulterated `s->firstc` value (to support future expansion), which probably also should be used in `get_histtype`

Reply:
Ok, why isn't `ccline.cmdfirstc = s->firstc` simply? Why make it 0 for '@'?

Reply:
Got it. This has to be 0 for both '@' and '#' because it is the first character shown on the command line.

So we keep it 0 for both
**OR** we can change all the `cmdfirstc == NUL` checks.